pkg/pool-stable/test/StablePool.test.ts

```diff
@@ -999,5 +1150,164 @@ describe('StablePool', function () {
+                : 'LPs can exit without changing the rate'
+            }`, async () => {
+              const totalBPT = await pool.totalSupply();
+              const totalBptBalance = await pool.balanceOf(lp);
+              const N = 4; // Number of sequential withdrawals
+              // Remove 1/(N+1) per step, so that we don't go down to zero (keeping rate valid)
+              const expectedAmountsOut = initialBalances.map((balance) => balance.div(N + 1));
+              let currentRate = await pool.getRate();
+              const bptBurnedPerExit = totalBptBalance.div(N + 1);
+
+              // Exit N times, ensuring we get the correct amount out, the tokens are burned, and the rate calculation remains valid
+              for (let i = 1; i <= N; i++) {
+                const result = await pool.recoveryModeExit({ from: lp, bptIn: bptBurnedPerExit });
+
+                // Protocol fees should be zero
+                expect(result.dueProtocolFeeAmounts).to.be.zeros;
+
+                // Balances are reduced by half because we are returning half of the BPT supply
+                expect(result.amountsOut).to.equalWithError(expectedAmountsOut, 0.001);
+
+                // Current BPT balance should have been reduced by half
+                expect(await pool.balanceOf(lp)).to.be.equalWithError(
+                  totalBptBalance.sub(bptBurnedPerExit.mul(i)),
+                  0.001
+                );
+
+                // Total supply should be reduced by half (ensure tokens are burned)
+                expect(await pool.totalSupply()).to.equalWithError(totalBPT.sub(bptBurnedPerExit.mul(i)), 0.000001);
+                expect(await pool.getRate()).to.equalWithError(currentRate, 0.0001);
+
+                if (intersperseTrades) {
+                  if (await pool.instance.invariantConverges()) {
+                    await pool.swapGivenIn({ in: 0, out: 1, amount: fp(100) });
+
+                    // Update the expected rate
+                    currentRate = await pool.getRate();
+                  } else {
+                    // Trade will fail
+                    await expect(pool.swapGivenIn({ in: 1, out: 0, amount: fp(1) })).to.be.revertedWith(
+                      'STABLE_INVARIANT_DIDNT_CONVERGE'
+                    );
+                  }
+                }
+              }
+            });
+          }
+
+          context('with invariant working', async () => {
+            LPsCanExit();
+            LPsCanExit(true); // with trades in the middle
+          });
+        });
+
+        context('when not in recovery mode', () => {
+          it('is not in recovery mode', async () => {
+            expect(await pool.inRecoveryMode()).to.be.false;
+          });
+
+          it('recovery mode exit is not available', async () => {
+            await expect(pool.recoveryModeExit({ from: lp, bptIn: fp(1) })).to.be.revertedWith('NOT_IN_RECOVERY_MODE');
+          });
+        });
+      });
```

Comment:
I think all of this is also out of scope, as this is BasePool behavior. We're not introducing any StablePool specific changes here, so there's nothing to test. The only recovery mode thing we should look at is that hte last invariant is updated at the end.